src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs

```diff
@@ -182,8 +182,8 @@ private static SyntaxNode CreateNewArgumentNullException(SyntaxGenerator factory
         {
             var identifier = factory.IdentifierName(parameter.Name);
             var nullExpr = factory.NullLiteralExpression();
-            var condition = factory.SupportsPatterns(semanticModel.SyntaxTree.Options)
-                ? factory.IsPatternExpression(identifier, factory.ConstantPattern(nullExpr))
+            var condition = factory.SyntaxGeneratorInternal.SupportsPatterns(semanticModel.SyntaxTree.Options)
```

Comment:
hrmm, why this approach?

Reply:
The alternative is making these public APIs, which would be out of scope for this pull request.